platform/core/src/utils/buildButtonCommands.ts

```diff
@@ -0,0 +1,45 @@
+import { ButtonProps, RunCommand } from '../types';
+
+const toArray = <T>(value?: T | T[]): T[] =>
+  Array.isArray(value) ? value : value != null ? [value] : [];
+
+export const buildButtonCommands = (
+  buttonProps: ButtonProps,
+  baseArgs: Record<string, unknown>,
+  { servicesManager, commandsManager }: AppTypes.Managers
+): Array<() => unknown> => {
+  const allCommands: Array<() => unknown> = [];
+  const seenCommands = new Set<string>();
+
+  // 1) normalize item-level commands
+  for (const command of toArray(buttonProps.commands as RunCommand)) {
+    const key = JSON.stringify({ cmd: command, v: (baseArgs as any)?.value ?? null, src: 'item' });
+    if (seenCommands.has(key)) continue;
+    seenCommands.add(key);
+    allCommands.push(() => commandsManager.run(command, baseArgs));
+  }
+
+  // 2) normalize option-level commands
+  for (const option of toArray(buttonProps.options)) {
+    if (!option || option.ignoreOnToolbarInteraction || !option.commands) continue;
+
+    const valueToUse = option.value;
+    for (const command of toArray(option.commands)) {
+      const commandOptions = {
+        ...option,
+        value: valueToUse,
+        options: buttonProps.options,
+        servicesManager,
+        commandsManager,
+      };
+      const key = JSON.stringify({ cmd: command, v: valueToUse, src: 'option' });
+
+      if (seenCommands.has(key)) continue;
+
+      seenCommands.add(key);
+      allCommands.push(() => commandsManager.run(command, commandOptions));
+    }
+  }
+
```

Comment:
I'm kind of confused by this overal piece of code. What are you trying to do?

Reply:
I just removed them as they are not required to solve the issue.

But what it was doing was: it would generate a snapshot of the command, its arguments and its origin as text and compare it. If for whatever reason there was another command with the exact same snapshot, it would simply ignore and skip it. But as I said, this is not necessary, it's was just being extra careful.